ocs_sample_library_preview/Asset/AssetRule.py

```diff
@@ -0,0 +1,97 @@
+import json
+
+
+class AssetRule(object):
+    """OCS Asset Rule definition"""
+
+    def __init__(self, id: str = None, name: str = None, description: str = None):
+        """
+        :param id: required
+        :param name: required
+        :param description: not required
+        """
+        self.Id = id
+        self.Name = name
+        self.Description = description
+
+    @property
+    def Id(self) -> str:
+        """
+        required
+        :return:
+        """
+        return self.__id
+
+    @Id.setter
+    def Id(self, value: str):
+        """"
+        required
+        :param value:
+        :return:
+        """
+        self.__id = value
+
+    @property
+    def Name(self) -> str:
+        """
+        required
+        :return:
+        """
+        return self.__name
+
+    @Name.setter
+    def Name(self, value: str):
+        """
+        required
+        :param value:
+        :return:
+        """
+        self.__name = value
+
+    @property
+    def Description(self) -> str:
+        """
+        not required
+        :return:
+        """
+        return self.__description
+
+    @Description.setter
+    def Description(self, value: str):
+        """
+        not required
+        :param value:
+        :return:
+        """
+        self.__description = value
+
+    def toJson(self):
```

Comment:
`toJson` and `fromJson` could use typehints,
`def toJson(self) -> str`
`def fromJson(self) -> AssetRule`
I think the latter requires `from __future__ import annotations` at the top of the file.

Double checking this: It doesn't look like we added those typehints in the existing model files. So I wouldn't say this is a requirement for this PR, could be done in a separate PR for the whole library.

Reply:
I agree with this comment, but we don't use type hints for these functions in the rest of the library (see SdsStream for example). Maybe this would be better as a separate WI?

Reply:
Yeah, agreed, I think it's fine to do as a separate WI.